### ISSUE TITLE
Unable to load app from namespace

### ISSUE THREAD

**Issue by dmitrybelyakov** [projectshift/shift-boiler#102]
Ever since moving our `app.py` from using `bootstrap.create_app(__name__)` to `bootstrap.create_app(app_name)` flask fails o bootstrap unless init file is present at `backend/__init__.py`

**Comment by dmitrybelyakov** [comment]
## Isolating/reproducing the issue

We'll slightly modify Flask quickstart app to create a following structure:

```
/quickstart
    /hello.py
```

And use the following code for the app:

```python
from flask import Flask

app_name = __name__
app = Flask(app_name)

@app.route('/')
def hello_world():
    return 'Hello, World!'
```

Now this works because `__name__` is `quickstart.hello`. Wich we can verify by running:

```
export FLASK_APP=quickstart.hello && flask run
```

Now if we change the app code to:

```python
from flask import Flask

app_name = 'quickstart'
app = Flask(app_name)

@app.route('/')
def hello_world():
    return 'Hello, World!'
```

It will fail unless we add an `__init__.py` file into our `/quickstart` directory like so:

```
/quickstart
    /hello.py
    /__init__.py
```

Now it will work. Also having __init__.py in project root helps makes it work too:


```
/__init__.py
/quickstart
    /hello.py
```

## Background on the issue

The culprit seems to be how Python treats directories containing `__init__.py` files, as in regular packages (with init) vs. namespace packages (without init).

  * [Regular packages](https://docs.python.org/3/reference/import.html#regular-packages)
  * [Namespace packages](https://docs.python.org/3/reference/import.html#namespace-packages) (A PEP 420 package which serves only as a container for subpackages.)

### Caveat

There is a caveat however - how we address the app module will depend on whether the `__init__.py` is present or not in userland code. For example:

  * Having an `__init__.py` (regular package): `quickstart` works, `quickstart.app` doesn't
  * Having *no* `__init__.py` (namespace package): is backwards - `quickstart` doesn't work but `quickstart` does

**Whether there is an `__init__.py` in userland code is outside of our control and depends on how useres want to structure their app.**


## Additional details

As we have seen, regular packages and namespace packages are treated differently. We can verify this by running `dir()` on them.

Namespace package:

```
['__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__']
```

Regular package:

```
['__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__']
```

So regular packages will have additiopnal properties on them:

  * `__builtins__`
  * `__cached__`
  * `__doc__`

They will both be of type `<class 'module'>` however.

Another dirrence is how `__path__` is treated, where for regular packages `__path__` will be a simple list and for namespace packages it will be an instance of `_NamespacePath`

**Comment by dmitrybelyakov** [comment]
Tis is now fixed and released as part of `0.10.0`
